AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginIntegrationTests/DataStoreHubEventsTests.swift

```diff
@@ -0,0 +1,70 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import XCTest
+
+import AmplifyPlugins
+import AWSPluginsCore
+
+@testable import Amplify
+@testable import AmplifyTestCommon
+@testable import AWSDataStoreCategoryPlugin
+
+@available(iOS 13.0, *)
+class DataStoreHubEventTests: HubEventsIntegrationTestBase {
+
+    /// - Given:
+    ///    - two models: Post, Comment
+    ///    - no pending MutationEvents in MutationEvent database
+    /// - When:
+    ///    - DataStore starts booting up
+    /// - Then:
+    ///    - subscriptionEstablished received, payload should be nil
+    ///    - syncQueriesStarted received, payload should be: {models: ["Post", "Comment"]}
+    ///    - outboxStatus received, payload should be {isEmpty: true}
```

Comment:
```
/// - Given:
    ///    - registered models from `TestModelRegistration` contains `Post` and `Comment`
    ///    - no pending MutationEvents in MutationEvent database // <- how do you ensure this?
    /// - When:
    ///    - DataStore starts booting up  // <- DataStore's remote sync engine is initialized
    /// - Then:
    ///    - `subscriptionEstablished` received, payload should be nil 
    ///    - `syncQueriesStarted` received, payload should be: {models: ["Post", "Comment"]} 
    ///    - outboxStatus received, payload should be {isEmpty: true}
```

Reply:
- for this one:
```
registered models from `TestModelRegistration` contains `Post` and `Comment`
```
I actually don't guarantee whether the order be `["Post", "Comment"]` or `["Comment", "Post"]`, or is it in this order for sure? So I should probably just say:
```
registered two models from `TestModelRegistration`
```
- and for this one:
```
no pending MutationEvents in MutationEvent database // <- how do you ensure this?
```
By not making any `Amplify.DataStore.save()`? 😂

Reply:
- I don't think the order matters, nor the models that are in there. i'd say just remove the mention of post and comment. `registered two models from `TestModelRegistration` sounds good
- other tests running that does do Save to the same local store and if terminated before the mutation events are enqueued and processed, then this test will run with mutation events in the database. i would say we should just assert for at least one outbox mutation event, and then if there are more that is okay. you can achieve this by having the `expecatation.overfufill = true`